### PR TITLE
4.x Remove Pimple

### PR DESCRIPTION
**Pull request by l0gicgate**
Removing Pimple as per discussion in #2585.

### CODE REVIEW

File: tests/AppTest.php [hunk 1]
```diff
@@ -39,8 +37,7 @@ public static function setupBeforeClass()
 
     public function testGetContainer()
     {
-        $pimple = new Pimple();
-        $container = new Psr11Container($pimple);
+        $container = new MockContainer();
```
Comment (akrabat) [inline tests/AppTest.php:40]: Why have a `MockContainer` object when we could be using PHPUnit's `createMock()` ?

Comment (l0gicgate) [reply tests/AppTest.php:40]: We need to have ArrayAccess implemented on the class unless you want to force usage of the `ContainerInterface::set()` method

Comment (akrabat) [reply tests/AppTest.php:40]: > We need to have ArrayAccess implemented on the class

Why? Slim doesn't set anything into the container anymore does it?

Comment (l0gicgate) [reply tests/AppTest.php:40]: I misread your question. We can mock a container for that specific test sure. We have a MockContainer though so why not use it

Comment (akrabat) [reply tests/AppTest.php:40]: Why do we have a `MockContainer` though? We shouldn't need it as we should be able to use PHPUnit's Mocking functionality for every situation where Slim's classes use a container. If we can't then we need to look at that particular Slim code.

**Comment by l0gicgate** [comment]
Will refactor with Mocks instead and re-open.

**Pull request closed without merging**